sycl/include/CL/sycl/detail/cg_types.hpp

```diff
@@ -27,13 +27,15 @@ namespace detail {
 class ArgDesc {
 public:
   ArgDesc(cl::sycl::detail::kernel_param_kind_t Type, void *Ptr, int Size,
-          int Index)
-      : MType(Type), MPtr(Ptr), MSize(Size), MIndex(Index) {}
+          int Index, int ElemSize = 0)
+      : MType(Type), MPtr(Ptr), MSize(Size), MIndex(Index),
+        MElemSize(ElemSize) {}
 
   cl::sycl::detail::kernel_param_kind_t MType;
   void *MPtr;
   int MSize;
   int MIndex;
+  int MElemSize;
```

Comment:
The patch should break ABI since this structure crosses library boundaries. Breaking ABI is not allowed right now.